routes/index.js

```diff
@@ -1,519 +1,565 @@
-    var accessToken = req.get('x-auth-access-token');
 
-    var challengePayload = fMassagePayload(req.body);
-    var cb = function(err, httpResponse, body) {
-        if (err) {
-            res.status(500).json(body);
-        } else {
-            var challenge = {};
-
-            try {
-                challenge = JSON.parse(body);
-                challenge.success = true;
-                challenge.challengeURL = config.TC_SITE + '/challenge-details/' + challenge.id +
-                    '/?type=develop&noncache=true'
-            } catch (e) {
-                challenge = {
-                    'error': body
-                };
-            }
-            res.json(challenge);
-        }
-    };
-
-    if (accessToken) {
-        console.log('I gots me an accessToken!', accessToken);
-        fCreateChallenges(accessToken, challengePayload, cb);
-    } else {
-        console.log('no accessToken. fetching first, then creating');
-        var user = (req.body && req.body.x_auth_username) ? req.body.x_auth_username : '';
-        var pass = (req.body && req.body.x_auth_password) ? req.body.x_auth_password : '';
-
-        fGetAccessToken(user, pass, function(err, httpResponse, body) {
-            if (err) {
-                console.log(err);
-                res.status(500).json(body);
-            } else {
-                accessToken = JSON.parse(body).x_auth_access_token;
-                console.log('I gots me an accessToken!', accessToken);
-                fCreateChallenges(accessToken, challengePayload, cb);
-            }
+/**
+ * GET /challenges - Gets the list of challenges.
+ * 
+ * INPUTS:
+ * - Authorization header with Bearer accessToken (required)
+ * - Query parameters in the same format expected by the TopCoder API (optional):
+ *   https://github.com/cwdcwd/topcoder-api-challenges/blob/master/docs/DefaultApi.md#challengesGet
+ * 
+ * OUTPUT: Array of challenges in JSON format in the response body
+ */
+router.get('/challenges', function(req, res, next) {
+    // Get the accessToken from the Authorization header
+    var accessToken = req.headers.authorization;
+    
+    // The accessToken is required
+    if (!accessToken) {
+        res.status(401).json({
+            error: 'No accessToken found in request header.'
         });
+        return;
     }
+    
+    // Remove the 'Bearer ' prefix
+    accessToken = accessToken.replace('Bearer ', '');
+    
+    // Get the options from the query parameters
+    var options = req.query || {};
+    
+    // Call the TopCoder API to get the list of challenges
+    fGetChallenges(accessToken, options, function(error, challenges) {
+        if (error) {
+            res.status(500).json({
+                error: error.message
+            });
+            return;
+        }
+        
+        // Send back the challenge list JSON
+        res.status(200).json(challenges);
+    });
 });
 
-router.post('/challenges/:service?', function(req, res, next) {
```

Comment:
@jmgasper this one should have stayed as it allows us to do both github and gitlab. I'll try and revert this chunk